nucliadb_vectors/src/vectors/data_point_provider/merge_worker.rs

```diff
@@ -52,28 +52,32 @@ impl Worker {
         let lock = directory::shared_lock(subscriber)?;
         let state: State = directory::load_state(&lock)?;
         std::mem::drop(lock);
-        if let Some(work) = state.current_work_unit() {
-            let work: Vec<_> = work
-                .iter()
-                .rev()
-                .map(|j| (state.delete_log(*j), j.id()))
-                .collect();
-            let new_dp = DataPoint::merge(subscriber, &work)?;
-            let ids: Vec<_> = work.into_iter().map(|(_, v)| v).collect();
-            std::mem::drop(state);
 
-            let report = self.merge_report(ids.iter().copied(), new_dp.meta().id());
-            let lock = directory::exclusive_lock(subscriber)?;
-            let mut state: State = directory::load_state(&lock)?;
-            state.replace_work_unit(new_dp);
-            directory::persist_state(&lock, &state)?;
-            std::mem::drop(lock);
-            info!("Merge on {subscriber:?}:\n{report}");
-            ids.into_iter()
-                .map(|dp| (subscriber, dp, DataPoint::delete(subscriber, dp)))
-                .filter(|(.., r)| r.is_err())
-                .for_each(|(s, id, ..)| info!("Error while deleting {s:?}/{id}"));
-        }
+        let Some(work) = state.current_work_unit().map(|work|
+            work
+            .iter()
+            .rev()
+            .map(|j| (state.delete_log(*j), j.id()))
```

Comment:
`j` stands for `Journal` right ? Since `j` is used in iteration (akin to `i`), it's a bit confusing and I'd suggest to rename that variable :cat:

Reply:
Done at [b5a8af7](https://github.com/nuclia/nucliadb/pull/455/commits/b5a8af788ee8915678e1d15695fc046e496ab20c)